File: practicas-lic/Diamante-X-Y-Z-8/java/A.java
package java;
public class A {
    public int x;
    public int z;

    public A(int x, int z) {
        this.x = x;
        this.z = z;
        System.out.println("Constructor de A: x=" + this.x + ", z=" + this.z);
    }

    public void incrementaX() {
        this.x++;
        System.out.println("A - incrementaX: x=" + this.x);
    }

    public void incrementaZ() {
        this.z++;
        System.out.println("A - incrementaZ: z=" + this.z);
    }
}

File: practicas-lic/Diamante-X-Y-Z-8/A.py
class A:
    def __init__(self, x, z):
        self.x = x
        self.z = z
        print(f"Constructor de A: x={self.x}, z={self.z}")
    def incrementaX(self):
        self.x = self.x + 1
        print(f"A - incrementaX: x={self.x}")
    def incrementaZ(self):
        self.z = self.z + 1
        print(f"A - incrementaZ: z={self.z}")

class B:
    def __init__(self, y, z):
        self.y = y
        self.z = z
        print(f"Constructor de B: y={self.y}, z={self.z}")
    def incrementaY(self):
        self.y = self.y + 1
        print(f"B - incrementaY: y={self.y}")
    def incrementaZ(self):
        self.z = self.z + 1
        print(f"B - incrementaZ: z={self.z}")

class D(A, B):  
    def __init__(self, x, y, z):
        A.__init__(self, x, z) 
        B.__init__(self, y, z)  
        print(f"Constructor de D: x={self.x}, y={self.y}, z={self.z}")
    def incrementaXYZ(self):
        self.incrementaX()
        self.incrementaY()
        self.incrementaZ() 
d_obj = D(1, 2, 3)
d_obj.incrementaX()
d_obj.incrementaY()
d_obj.incrementaZ() 
d_obj.incrementaXYZ()

print(f"Valores finales en D: x={d_obj.x}, y={d_obj.y}, z={d_obj.z}")

print(D.mro()) 